source/speech/sayAll.py

```diff
@@ -15,92 +18,278 @@
 import queueHandler
 import winKernel
 
-from speech.commands import CallbackCommand, EndUtteranceCommand
-
+from .commands import (
+	# Commands that are used in this file.
+	LangChangeCommand,
+	EndUtteranceCommand,
+	CallbackCommand,
+)
+
+from .types import (
+	SpeechSequence,
+	logBadSequenceTypes,
+	GeneratorWithReturn,
+	_flattenNestedSequences,
+)
+
+from typing import (
+	Optional,
+	Generator,
+	Callable,
+	Literal,
+	TYPE_CHECKING,
+)
+
+from .speech import (
+	getTextInfoSpeech,
+	speak,
+	SpeakTextInfoState,
+	speakObject,
+)
+
+if TYPE_CHECKING:
+	import NVDAObjects
 
 CURSOR_CARET = 0
 CURSOR_REVIEW = 1
+CURSOR = Literal[CURSOR_REVIEW, CURSOR_CARET]
+SayAllHandler = None
+
+
+def initalize():
+	global SayAllHandler
+	SayAllHandler = _SayAllHandler()
+
+
+def _yieldIfNonEmpty(seq: SpeechSequence):
+	"""Helper method to yield the sequence if it is not None or empty."""
+	if seq:
+		yield seq
+
+
+class SpeechWithoutPauses:

```

Comment:
Did you consider putting this in a separate module?

Reply:
I recall putting this in the same module being suggested as this is the only place it is used, I'm happy to move this if you think it should go somewhere else.

Reply:
I think that gives us more room for future designs, and doesn't cost much now or in maintenance.